src/main/java/org/commonjava/util/partyline/spi/JoinableFilesystem.java

```diff
@@ -0,0 +1,33 @@
+/**
+ * Copyright (C) 2015 Red Hat, Inc. (nos-devel@redhat.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.util.partyline.spi;
+
+import org.commonjava.util.partyline.callback.StreamCallbacks;
+import org.commonjava.util.partyline.lock.local.LocalLockManager;
+import org.commonjava.util.partyline.lock.local.LocalLockOwner;
+import org.commonjava.util.partyline.lock.local.ReentrantOperationLock;
+
+import java.io.File;
+import java.io.IOException;
+
+public interface JoinableFilesystem
+{
+    JoinableFile getFile( File file, LocalLockOwner lockOwner, StreamCallbacks callbacks, boolean doOutput,
+                          ReentrantOperationLock opLock )
+            throws IOException, InterruptedException;
+
+    LocalLockManager getLocalLockManager();
```

Comment:
We need an exists() method here, and we need to expose that as public API in the Partyline class too. We might need an exists() method in FileTree if we don't keep the JoinableFilesystem in the Partyline class directly.

When we move to a clustered storage system, new File("foo").exists() won't work for us anymore. This is what we're using currently in Galley, so we need to pave the way for that change.

Let's put this as lower priority and I'll add a new JIRA story for it. We can always do this as cleanup work after the initial implementation.